### ISSUE TITLE
Idea: Automated filtering

### ISSUE THREAD

**Issue by evanp** [swicg/activitypub-trust-and-safety#28]
Automated filtering with machine learning models are an important part of trust and safety on social networks.

Techniques like [naive bayesian filtering](https://en.wikipedia.org/wiki/Naive_Bayes_spam_filtering) have been used for detecting spam and other unwanted content in email. [Akismet](https://akismet.com/) is a blog spam filtering system. [ActivitySpam](https://gitlab.com/evanp/activityspam) is a server for spam detection in Activity Streams 1.0 JSON.

Investigating how automated filtering fits into the ActivityPub ecosystem is a good use case for this working group.

**Comment by ThisIsMissEm** [comment]
I'm not sure if this would be suitable for our initial report nor for our first round of work. Unless there's some recommendation text we can come up with for this for the initial report that clarifies what AP already non-normatively specifies?

I'm going to label this as a future issue for now.

**Comment by ThisIsMissEm** [comment]
I think this would be most likely implemented via the [FASP protocol](https://github.com/mastodon/fediverse_auxiliary_service_provider_specifications/), which itself uses ActvityPub / ActivityStreams 2 semantics. Additionally it could be potentially augmented with or aligned with the concept of "third-party annotations" which we're starting to define for content labeling but also for community notes. 

This would be like some sort of Annotation object created by an Actor, and you'd Follow that Actor to receive their annotations.

There's still a fair few questions here to be answered, particularly around trying to figure out exactly how to do what we want with Web Annotations, if that's even the right tool for the job.

Given this, would you be happy with this being closed @evanp ?

**Comment by trwnh** [comment]
I wonder if this can be handled as part of #33

**Comment by ThisIsMissEm** [comment]
@trwnh yes, but this needs more information first. The topic area here is extremely vast, so there's perhaps not a heap we can normatively define here.

**Comment by ThisIsMissEm** [comment]
Discussed in the interim meeting on 19th August 2026, closing as unactionable at this time.

**Comment by renchap** [comment]
We (Mastodon) will be working on something related starting in October, as part of our [contract with STA](https://blog.joinmastodon.org/2026/04/sovereign-tech-agency-funding/).

Our goal is to create what we currently call a "Automated Content Scanning FASP", to allow fediverse administrators to pick one (or multiple) services (running outside of the Mastodon software, could be a 3rd party, could be self-hosted) to scan the content going through their instance and emit recommendations or actions based on it. I will do my best to communicate about it once we have something to share.

**Comment by ThisIsMissEm** [comment]
@renchap yeah, though the FASP protocol is outside the remit of this task force. I'd recommend talking with @cassidyjames about this — the key to automated moderation systems is data over time, though different systems want different shapes, event structured data is usually a good starting point.
